### ISSUE TITLE
reward/chosen

### ISSUE THREAD

**Issue by zhangguoxin1** [princeton-nlp/SimPO#42]
![image](https://github.com/user-attachments/assets/1231e002-8c11-4251-bba2-1fb02a067007)
Hi! 
    I am fine-tuning LLaMA3 on the hh-rlhf dataset using SimPo and noticed that the reward/chosen reward is decreasing. Is this reasonable?

**Comment by zhangguoxin1** [comment]
I expected the reward/chosen to increase, but since the goal of SimPo is to maximize the difference between reward/chosen and reward/rejected, it is acceptable for reward/chosen to decrease to a certain extent. However, the extent of the decrease in reward/chosen seems a bit large compared to reward/chosen - reward/rejected.

**Comment by zhangguoxin1** [comment]
@danqi

**Comment by yumeng5** [comment]
Hi,

Yes, this is reasonable. The reward margin should increase but the reward on chosen responses may slightly decrease (and the reward on rejected decreases more rapidly). In general, we don't want the reward on chosen to decrease too much (as that implies the likelihood of chosen responses is decreasing), and you may use a larger `beta` or a lower learning rate to mitigate the decrease of reward on chosen responses.

Best,
Yu

**Comment by zhangguoxin1** [comment]
get it!

Thanks for the quick reply.

**Comment by zhangguoxin1** [comment]
Hi,
    I used Simpo in my task with qwen2_7B, but the model generated repeated sentences and pre-trained data. The parameters are as follows:
```bash
pref_beta: 2.5
simpo_gamma: 1.0
learning_rate: 1.0e-6
num_train_epochs: 3.0
```

![image](https://github.com/user-attachments/assets/a41f6a0f-03fc-4b2f-9e56-e6f8640dd086)

**Comment by xiamengzhou** [comment]
@zhangguoxin1 
I think you should be using Qwen2-7B-Instruct rather than Qwen2-7B if you only running PO? Also I'd suggest that you use online data rather offline data that is generated by other models.

**Comment by yumeng5** [comment]
Hi @zhangguoxin1 

In addition to the suggestions by Mengzhou, you may try the following as well:

- decrease the learning rate (we usually start learning rate search around 5e-7)
- reduce the number of training epochs (we generally train the model for only one epoch)

Best,
Yu
